digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -160,7 +164,50 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
     case ai: AdditionalInformation => JString(ai.toString)
   }))
 
-  protected implicit val jsonFormats: Formats = DefaultFormats + DateTimeSerializer + LinkGeomSourceSerializer + SideCodeSerializer + TrafficDirectionSerializer + LinkTypeSerializer + DayofWeekSerializer + AdministrativeClassSerializer + WidthLimitReasonSerializer + AdditionalInfoClassSerializer + PointAssetSerializer
+  case object NewIncomeLaneSerializer extends CustomSerializer[NewIncomeLane](format =>
+    ( {
+      case jsonObj: JObject =>
+        val id = (jsonObj \ "id").extract[Long]
+        val startMeasure = (jsonObj \ "startMeasure").extract[Double]
+        val endMeasure = (jsonObj \ "endMeasure").extract[Double]
+        val municipalityCode = (jsonObj \ "municipalityCode").extract[Long]
+        val isExpired = (jsonObj \ "isExpired").extractOrElse[Boolean](false)
+        val isDeleted = (jsonObj \ "isDeleted").extractOrElse[Boolean](false)
+        val properties = (jsonObj \ "properties").extract[Seq[LaneProperty]]
+
+        NewIncomeLane(id, startMeasure, endMeasure, municipalityCode,isExpired, isDeleted, LanePropertiesValues(properties))
+    },
+      {
+        case tv : NewIncomeLane => Extraction.decompose(tv)
+      }))
+
+  case object LanePropertySerializer extends CustomSerializer[LaneProperty](format =>
+    ({
+      case jsonObj: JObject =>
+        val publicId = (jsonObj \ "publicId").extract[String]
+        val values = (jsonObj \ "values").extract[Seq[LanePropertyValue]]
+
+        LaneProperty(publicId, values)
+    },
+      {
+        case tv : LaneProperty => JObject(JField("publicId", JString(tv.publicId)), JField("values", Extraction.decompose(tv.values)))
+      }))
+
+  case object LanePropertyValueSerializer extends CustomSerializer[LanePropertyValue](format =>
+    ({
+      case jsonObj: JObject =>
+        val value = (jsonObj \ "value").extract[Any]
+
+        LanePropertyValue(value)
+    },
+      {
+        case tv : LanePropertyValue => JObject(JField("value", JString(tv.value.toString)))
+      }))
```

Comment:
This part of the code needs to be deleted and done like the other old assets.
Do it like linear already extract the info on "post("/linearassets") {" for example.
This is the idea, not creating a Serializer for all.

Reply:
Done